File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        for i in range(0,len(nums)):
            for j in range(0,len(nums)):
                if i!=j:
                    if nums[i]+nums[j]==target:
                        return [i,j]
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,vector<int>>mp;
        for(int i=0;i<nums.size();i++){
            mp[nums[i]].push_back(i);
        }
        for(auto &i:mp){
            int rest=target-(i.first);
            if(rest==i.first && (i.second).size()>=2){
              return i.second;
                
            }
            else if(rest!=i.first && mp.find(rest)!=mp.end()){
                return {i.second[0],mp[rest][0]};
            }
        }
        return {0,0};
    }
};